ledger/participant-integration-api/src/main/scala/platform/store/cache/StateCache.scala

```diff
@@ -16,11 +16,16 @@ import scala.concurrent.{ExecutionContext, Future}
 /** This class is a wrapper around a Caffeine cache designed to handle
   * correct resolution of concurrent updates for the same key.
   */
-private[platform] case class StateCache[K, V](cache: Cache[K, V], registerUpdateTimer: Timer)(
-    implicit ec: ExecutionContext
-) {
+private[platform] case class StateCache[K, V](
+    initialCacheIndex: Long,
+    cache: Cache[K, V],
+    registerUpdateTimer: Timer,
+)(implicit ec: ExecutionContext) {
   private val logger: ContextualizedLogger = ContextualizedLogger.get(getClass)
   private[cache] val pendingUpdates = mutable.Map.empty[K, PendingUpdatesState]
```

Comment:
Given that we synchronize on this is `TrieMap` worth considering ?

Reply:
no, because we do some processing between the fetch and update so need to lock map.